includes/class-sensei-frontend.php

```diff
@@ -41,7 +41,7 @@ public function __construct () {
 		add_action( 'sensei_lesson_archive_lesson_title', array( $this, 'sensei_lesson_archive_lesson_title' ), 10 );
 
 		// 1.2.1
-		add_action( 'wp_head', array( $this, 'sensei_complete_lesson' ), 10 );
```

Comment:
You could also hook into `wp` action and take advantage of the loop (instead of passing `lesson_id`).